tests/distributions/test_lkj.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import absolute_import, division, print_function
+
+import pytest
+import torch
+from torch.distributions import biject_to, transform_to
+from pyro.distributions.lkj import (corr_cholesky_constraint, CorrLCholeskyTransform)
+from tests.common import assert_tensors_equal
+
+
+@pytest.mark.parametrize("value_shape", [(1, 1), (3, 3), (5, 5)])
+def test_constraint(value_shape):
+    value = torch.randn(value_shape).clamp(-2, 2).tril()
+    value.diagonal(dim1=-2, dim2=-1).exp_()
+    value = value / value.norm(2, dim=-1, keepdim=True)
+
+    #assert_tensors_equal(corr_cholesky_constraint.check(value), torch.ones(value_shape[:-2], dtype=value.dtype))
+    assert (corr_cholesky_constraint.check(value) == 1).all()
+
+
+def _autograd_log_det(ys, x):
+    # computes log_abs_det_jacobian of y w.r.t. x
+    return torch.stack([torch.autograd.grad(y, (x,), retain_graph=True)[0]
+                        for y in ys]).det().abs().log()
+
+
+@pytest.mark.parametrize("y_shape", [(1,), (3, 1), (6,), (1, 6), (2, 6)])
+def test_unconstrained_to_corr_cholesky_transform(y_shape):
+    transform = CorrLCholeskyTransform()
+    y = torch.empty(y_shape).uniform_(-4, 4).requires_grad_()
+    x = transform(y)
+
+    # test codomain
+    assert (transform.codomain.check(x) == 1).all()
+
+    # test inv
+    y_prime = transform.inv(x)
+    assert_tensors_equal(y, y_prime, prec=1e-4)
+
+    # test domain
+    assert (transform.domain.check(y_prime) == 1).all()
+
+    # test log_abs_det_jacobian
+    log_det = transform.log_abs_det_jacobian(y, x)
+    assert log_det.shape == y_shape[:-1]
+    if y_shape == ():
+        assert_tensors_equal(torch.autograd.grad(y, (x,), retain_graph=True)[0].abs().log(), log_det)
```

Comment:
Correct... The tests were written by @fehiepsi.  I didn't want to alter/delete more of his code than necessary, as a courtesy, and because I figured that if he added it, he must have had something in mind.  (I'm not sure why we care about an empty shape...)  Would you prefer for me to add a test for this situation or to take out the test?

Reply:
`autograd.grad` works for a transform from `x` to `y`, not from . It should be fine if you change back to my [original notations](https://github.com/elbamos/pyro/pull/4/files#diff-45451189a07bb3d85b0b27895c9328cdR52).

Reply:
In addition, here and there you add prec=1e-4. Pyro's test uses DoubleTensor so I think that adding precision is unnecessary given that the transform is just a combination of some normal functions.

Reply:
This is the reason why I have to fix the code while adding tests. Without fixing (I mean by just adding tests alone), you might have no clue if the code is wrong or the test is wrong like this case,... You can freely modify the fixed code to your favourite, as long as the tests pass. This might cause a big issue when the test and the code both are wrong, so we need to review. But generally, I think that reviewing the test is easier than reviewing the code.

Reply:
@fehiepsi The tests at issue don't actually test anything, and they were never called by your code.  

I changed your notation so it would be identical to the notation used in both the Stan source code and in the original papers implemented by Stan (which was the same notation I had used throughout). 

Ok, I will remove the unnecessary tests.

Reply:
I see some differences between that version and the version you put here. There, the condition is `if len(x_shape) == 1:` (so it is `len(y_shape) == 1` with your notation). Here the condition is `if y_shape == ():`, so it will be ignored. You can remove the test but I recommend to replace by another test for Jacobian to see if the transform `log_abs_det_jacobian` method is implemented correctly. Here your transform has `event_dim=1` so it is required to test for input with `ndim=1`. If you test for some transforms else with `event_dim=0` (e.g. AffineTransform, TanhTransform) then testing for constant input is enough.